CredentialProvider.Microsoft/CredentialProviders/Vsts/IAuthUtil.cs

```diff
@@ -145,10 +152,10 @@ protected virtual async Task<HttpResponseHeaders> GetResponseHeadersAsync(Uri ur
                     httpClient.DefaultRequestHeaders.UserAgent.Add(userAgent);
                 }
 
-
                 logger.Verbose($"GET {uri}");
-                using (var response = await httpClient.SendAsync(request, cancellationToken))
+                using (var response = await httpClient.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, cancellationToken))
```

Comment:
I'm not sure if reading only the headers and then disposing the response will make the connection non-reusable.  If this causes the connection to close when it would otherwise be re-used later (with auth), then it might make sense to just read the whole thing here.

Reply:
oh goodness and we were newing up a new HttpClient each time :(

Reply:
Yes yes, HttpClient usage is quite terrible and would love to fix that... can't spend all my time refactoring the code ;). Will be slowly morphing this into the modern tech world, and using IHttpClientFactory and the like is part of that.

Reply:
> I'm not sure if reading only the headers and then disposing the response will make the connection non-reusable. If this causes the connection to close when it would otherwise be re-used later (with auth), then it might make sense to just read the whole thing here.

We would never reuse anyway, we never actually hit the pkgs endpoint for anything else, only authentication endpoints. And for creating session tokens that's a different host.